src/schema/schema_create.c

```diff
@@ -573,13 +573,14 @@ __create_table(WT_SESSION_IMPL *session,
 	if (!WT_PREFIX_SKIP(tablename, "table:"))
 		return (__wt_unexpected_object_type(session, name, "table:"));
 
-	if ((ret = __wt_schema_get_table(session,
-	    tablename, strlen(tablename), false, &table)) == 0) {
+	if ((ret = __wt_schema_get_table(
+	    session, tablename, strlen(tablename), false, 0, &table)) == 0) {
 		if (exclusive)
 			WT_ERR(EEXIST);
 		exists = true;
 	}
-	WT_ERR_NOTFOUND_OK(ret);
+	if (ret != ENOENT)
```

Comment:
I've tried to explain inline: if the metadata for a table doesn't exist, the `WT_NOTFOUND` from the metadata search causes `ENOENT` to be returned.  That's inherited from the btree code, and we do the same conversion at a higher level, but I'm not sure it's okay to change the low-level code without more thought so I'm just dealing with it here.